### PR TITLE
Ivan Filipenkov

### PR DESCRIPTION
**Pull request by vanutp**

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -4,22 +4,45 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short [] = True
+short [_] = True
+short [_, _] = True
+short _ = False
```
Comment (kajigor) [inline Main.hs:10]: There is a way to implement this in 2 lines. Can you do it?

File: Main.hs [hunk 2]
```diff
@@ -4,22 +4,45 @@ import Control.Monad (unless)
...
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely (_:_:c:_) = c == 14
+lovely x = short x
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = []
```
Comment (kajigor) [inline Main.hs:17]: ?

File: Main.hs [hunk 3]
```diff
@@ -4,22 +4,45 @@ import Control.Monad (unless)
...
+        "Jupiter" -> 11.862615
+        "Saturn" -> 29.447498
+        "Uranus" -> 84.016846
+        "Neptune" -> 164.79132
+        "Pluto" -> error "Pluto is not a planet"
+        _ -> error "Unknown planet"
 
 isLeapYear :: Int -> Bool
-isLeapYear year = undefined 
+isLeapYear year = (year `mod` 4 == 0) && (year `mod` 100 /= 0 || year `mod` 400 == 0)
```
Comment (kajigor) [inline Main.hs:45]: What about negative years?

**Review by kajigor — CHANGES_REQUESTED** [review]

**Comment by alex28sh** [comment]
ok

**Pull request closed without merging**